### PR TITLE
Switch to Biomejs

### LINKED ISSUE #1195: Switch to Biome
I'd like us to consider switching to [Biome](https://biomejs.dev/). Biome provides both a formatter and a linter, and is super fast compared to ESLint and Prettier. I've been using Biome for some other projects and I've been really pleased with it. It has a really sane default configuration, is easy to configure, [has good editor/IDE support](https://biomejs.dev/guides/editors/first-party-plugins/), and I find its output/suggestions much more helpful.

Its [Differences with Prettier](https://biomejs.dev/formatter/differences-with-prettier/) page is also a good read.

Below are some timing results (using [GNU time](https://www.gnu.org/software/time/)) on the Samf4 project on master, using the default Biome config.

## ESLint

```
root@e72eafc92be7:/app# time yarn run eslint:check

real	0m6.858s
user	0m9.880s
sys	0m1.168s
```

**Time to run: ~6s**

## Biome lint

```
root@e72eafc92be7:/app# time yarn run biome lint .

***truncated output***

real	0m0.668s
user	0m0.841s
sys	0m0.210s
```

**Time to run: ~0.6s**

## Biome format

```
root@e72eafc92be7:/app# time yarn run biome format .

***truncated output***

real	0m0.710s
user	0m0.747s
sys	0m0.271s
```

**Time to run: ~0.7s**

## Biome check (same as analyze+lint+format in one step)

```
root@e72eafc92be7:/app# time yarn run biome check .

***truncated output***

real	0m0.733s
user	0m0.953s
sys	0m0.270s
```

**Time to run: ~0.7s**

### PR DESCRIPTION
**Pull request by robines**
Closes #1195 

See the justification for switching from ESLint+Prettier to Biome in the linked issue.

# Usage

Biome consists of three main tools:

* Analyzer (currently only provides [import sorting](https://biomejs.dev/analyzer/import-sorting/))
* Formatter
* Linter

It's possible to run these tools individually, or all of them at once. The simplest way is obviously to run all at once, which you can do with `yarn biome:check` and `yarn biome:fix` to check and fix Biome issues respectively.

Other self-explanatory commands which are available are: `yarn format:check`/`yarn format:fix` and `yarn lint:check`/`yarn lint:fix`. The linter distinguishes between **safe** and **unsafe** code changes. Running `yarn lint:fix` will only apply **safe** code fixes. To apply both safe and unsafe fixes, run `yarn lint:fix-unsafe`. It's advised to manually review unsafe changes, as they may break things.

# Upgrade instructions

I reccommend [installing the Biome extension](https://biomejs.dev/guides/editors/first-party-plugins/) for your editor/IDE to highlight issues and let you easily fix them.

## Docker

Rebuild the frontend image with `docker compose build frontend`

## Local

Simply running `yarn` should do the trick.

### CODE REVIEW

**Comment by magsyg** [comment]
Damn var mange commits as

**Comment by robines** [comment]
@magsyg Yeah, lett å reverte enkelte endringer på den måten

**Review by Snorre98 — COMMENTED** [review]
Rust mentioned! 🦀🦀🦀🦀
@robines æt meg når du har merget og resolved master

**Comment by robines** [comment]
@Snorre98 Tenker å vente med denne PRen til de fleste andre PRer er inne. Om den merges må nesten alle andre PRer fikses opp i, som er slitsomt.

File: .vscode/settings.emil.json [hunk 1]
```diff
@@ -38,15 +38,8 @@
...
   // ================================
-  "eslint.enable": true,
-  "eslint.format.enable": true,
   "html.format.enable": false,
-  "prettier.configPath": "frontend/.prettierrc.js",
-  "prettier.ignorePath": "frontend/.prettierignore",
-  "editor.defaultFormatter": "esbenp.prettier-vscode",
-  "eslint.workingDirectories": ["frontend"],
   "[javascript][javascriptreact][typescript]": {
-    "editor.defaultFormatter": "esbenp.prettier-vscode",
```
Comment (emilte) [inline .vscode/settings.emil.json:49]: Dette er jo min fil da :P

Comment (robines) [reply .vscode/settings.emil.json:49]: Bare å reverte den hvis du vil, tenkte bare det var greit å holde alt i prosjektet oppdatert

Comment (emilte) [reply .vscode/settings.emil.json:49]: no worries

**Pull request merged**